pyccl/pk2d.py

```diff
@@ -156,20 +178,29 @@ def pk_from_model(Pk2D, cosmo, model):
             pk2d.psp, status = ret
 
         check(status, cosmo)
-        pk2d.has_psp = True
         return pk2d
 
     @classmethod
-    def apply_halofit(Pk2D, cosmo, pk_linear):
+    @functools.wraps(from_model)
+    def pk_from_model(cls, cosmo, model):
```

Comment:
did we lose the docstring? Or does `wraps` copy it?

Reply:
yes, exactly, `wraps` preserves it